store/sqlstore/channel_store.go

```diff
@@ -631,10 +631,10 @@ func (s SqlChannelStore) saveChannelT(transaction *gorp.Transaction, channel *mo
 }
 
 // Update writes the updated channel to the database.
-func (s SqlChannelStore) Update(channel *model.Channel) (*model.Channel, *model.AppError) {
+func (s SqlChannelStore) Update(channel *model.Channel) (*model.Channel, error) {
 	transaction, err := s.GetMaster().Begin()
 	if err != nil {
-		return nil, model.NewAppError("SqlChannelStore.Update", "store.sql_channel.update.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrap(err, "begin_transaction")
 	}
 	defer finalizeTransaction(transaction)
 
```

Comment:
Let's change `updateChannelT` also to return `error` ? Only the 
```
if err := channel.IsValid(); err != nil {
	return nil, err
}
```
part would still return an AppError as is. But we can convert the rest of them to the proper errors.

Reply:
Ok this is done